enhancements/network/ovn-on-hypershift.md

```diff
@@ -0,0 +1,271 @@
+---
+title: OVN on Hypershift
+authors:
+  - "@russellb
+  - "@squeed"
+reviewers:
+  - "@dcbw"
+  - @danwinship
+  - @tssurya
+  - @numans
+approvers:
+  - TBD
+creation-date: 2022-02-01
+last-updated: 2022-02-08
+tracking-link: 
+  - "https://issues.redhat.com/browse/SDN-2589"
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+---
+
+
+# OVN on Hypershift
+
+
+## Summary
+
+[Hypershift](https://hypershift-docs.netlify.app/) is a new deployment model for OpenShift which removes the requirement that clusters be "self-hosted". Rather, a given cluster's control plane actually resides as pods on another "management" cluster.
+
+Adapting the OVN networking plugin to the Hypershift model requires some careful design. Specifically, OVN provides its own control plane and datastore, all of which will be hosted external to the target cluster.
+
+Thus, the OpenShift OVN deployment model must be updated to support split control planes. Likewise, the Network Operator must be able to manage components on both the control plane as well as the target cluster.
+
+## Motivation
+
+We will elide the full motivation for hosted control planes (i.e. the Hypershift model). However, hosting the OVN control plane off-cluster has the same advantage as hosting the Kubernetes control plane:
+- potentially higher reliability
+- no special master nodes.
+- no chicken-egg problem while bootstrapping
+- More available features (e.g. volumes, stateful sets)
+- etc.
+
+The most significant motivation is the availability of StatefulSets for the OVN databases. In OpenShift,
+we use the ovsdb raft replication mode to provide high availability on master nodes. Managing this
+manually is error prone, especially in the case of lost masters. However, self-hosted OVN cannot use
+StatefulSets, since they require a functioning network. OVN on Hypershift does not suffer this
```

Comment:
I think the main benefit here is using persistent storage for the OVN databases.  Today they sit on ephemeral storage, which is OK because kube is the source of truth and the DBs can be rebuilt from scratch as needed.  It'd be better to avoid unnecessary database rebuilds though, especially as clusters get larger and/or more dense.

Reply:
FYI, ovn-k currently uses hostpath volumes, so we don't lose the database in general. However, the real bother is that we can't add or replace master nodes without manual intervention.

Reply:
ack thanks both,